torch/nn/modules/loss.py

```diff
@@ -297,6 +297,76 @@ def forward(self, log_input: Tensor, target: Tensor) -> Tensor:
                                   eps=self.eps, reduction=self.reduction)
 
 
+class GaussianNLLLoss(_Loss):
+    r"""Gaussian negative log likelihood loss.
+
+    The targets are treated as samples from Gaussian distributions with
+    expectations and variances predicted by the neural network. For a
+    1-dimensional ``target`` tensor modelled as having Gaussian distribution
+    with expectation ``input`` and positive variance ``var``, the loss is:
+
+    .. math::
```

Comment:
I feel like it would be nice to have only a single formula here.  In particular, isn't this one a special case of the one below?
Similarly for the third one, can you just say that if var is a scalar, it is broadcasted to be the same size as the  input?

Reply:
Agreed, I'll tidy this up.